ir/src/lib.rs

```diff
@@ -191,4 +196,21 @@ impl AirIR {
     pub fn constraint_graph(&self) -> &AlgebraicGraph {
         self.constraints.graph()
     }
+
+    // --- HELPER FUNCTIONS -----------------------------------------------------------------------
+
+    /// Checks whether auxiliary columns have been declared
+    fn validate_random_values(source: &Vec<SourceSection>) -> Result<(), SemanticError> {
+        for section in source {
+            if let SourceSection::Trace(trace) = section {
+                if trace.aux_cols.is_empty() {
+                    return Err(SemanticError::InvalidUsage(
+                        "Random Values section cannot be declared without aux trace columns"
+                            .to_string(),
+                    ));
+                }
+            }
+        }
+        Ok(())
+    }
```

Comment:
We probably want to use the source `validator` for this, so lets leave it for a future PR.